code/drasil-gool/GOOL/Drasil/LanguageRenderer/CppRenderer.hs

```diff
@@ -2610,15 +2610,14 @@ cpphClass :: Label -> CppHdrCode ParentSpec ->
 cpphClass n ps vars funcs pub priv = let 
   pubs = cpphVarsFuncsList Pub vars funcs
   privs = cpphVarsFuncsList Priv vars funcs
-  indLi :: Doc -> Doc -> [Doc]
-  indLi pb pr
-    | isEmpty pb = [RC.scope priv <> colon, indent pr]
-    | isEmpty pr = [RC.scope pub <> colon, indent pb]
-    | otherwise   = [RC.scope pub <> colon, indent pb, blank,
-      RC.scope priv <> colon, indent privs]
+  ifEmptyPubs = emptyIfEmpty pubs
+  ifEmptyPrivs = emptyIfEmpty privs
+  indLi = [ifEmptyPubs (RC.scope pub <> colon), ifEmptyPubs (indent pubs),
+          ifEmptyPubs (ifEmptyPrivs blank),
+          ifEmptyPrivs (RC.scope priv <> colon), ifEmptyPrivs (indent privs)]
```

Comment:
Here was what the naive solution looked like 
```haskell
cpphClass n ps vars funcs pub priv = let 
  pubs = cpphVarsFuncsList Pub vars funcs
  privs = cpphVarsFuncsList Priv vars funcs
  indLi = [emptyIfEmpty pubs (RC.scope pub <> colon), 
          emptyIfEmpty  pubs (indent pubs),
          emptyIfEmpty  pubs (emptyIfEmpty privs blank),
          emptyIfEmpty privs (RC.scope priv <> colon), 
          emptyIfEmpty privs (indent privs)] ............
```
This takes the same number of lines of code, is easier to read, but perhaps looks more repetitive. If you would like this version I could use it.

You may be looking at the stable version and noticing that there are still blank lines. 

for example in Projectile.hpp
```C++
class Constants {
    public:
        static const double g_vect;
        static const double epsilon;
        
};
```
However I believe that this is because the public code probably has some kind of empty line. The reason I say this is 1. If you put your cursor on the empty line it is indented. 2. if you edit the code and remove everything except for the public code you still see this empty line.

Reply:
Good catch about the blank line being indented. I bet it's the `cpphVarsFuncsList` function that is adding the blank line (which is not related to your changes). It's probably a quick fix. That might be something you are interested in doing (in a separate PR).

Reply:
I'm fine with the repetition in this code. That can always be optimized later.